pylegend/core/tds/pandas_api/frames/functions/filter.py

```diff
@@ -0,0 +1,214 @@
+            items: PyLegendOptional[PyLegendList[str]],
+            like: PyLegendOptional[str],
+            regex: PyLegendOptional[str],
+            axis: PyLegendOptional[PyLegendUnion[str, int, PyLegendInteger]],
+    ) -> None:
+        self.__base_frame = base_frame
+        self.__items = items
+        self.__like = like
+        self.__regex = regex
+        self.__axis = 1 if axis is None else axis
+
+    def __get_desired_columns(
+            self, col_names: PyLegendSequence[str]
+    ) -> PyLegendSequence[str]:
+        if self.__items is not None:
+            return self.__items
+        elif self.__like is not None:
+            return [col for col in col_names if self.__like in col]
+        elif self.__regex is not None:
+            regex_pattern = re.compile(self.__regex)
+            return [col for col in col_names if regex_pattern.search(col)]
+
+        return []
+
+    def to_sql(self, config: FrameToSqlConfig) -> QuerySpecification:
+        base_query = self.__base_frame.to_sql_query_object(config)
+        db_extension = config.sql_to_string_generator().get_db_extension()
+        should_create_sub_query = (
+                base_query.select.distinct
+                or (len(base_query.orderBy) > 0)
+                or (len(base_query.groupBy) > 0)
+                or (base_query.where is not None)
+        )
+        base_query = (
+            create_sub_query(base_query, config, "root")
+            if should_create_sub_query
+            else copy_query(base_query)
+        )
+
+        col_names = [c.get_name() for c in self.__base_frame.columns()]
+        quotes = db_extension.quote_character()
+
+        def _is_already_quoted(identifier: str) -> bool:
+            return identifier.startswith(quotes) and identifier.endswith(quotes)
+
+        desired_columns = [
+            SingleColumn(
+                alias=config.sql_to_string_generator()
+                .get_db_extension()
+                .quote_identifier(col),
+                expression=QualifiedNameReference(
+                    QualifiedName([
+                        f"{quotes}root{quotes}",
+                        col if _is_already_quoted(col) else f"{quotes}{col}{quotes}"
+                    ])
+                ),
+            )
+            for col in self.__get_desired_columns(col_names)
+        ]
+
+        base_query.select = Select(distinct=False, selectItems=cast(List[SelectItem], desired_columns))
+        return base_query
+
+    def to_pure(self, config: FrameToPureConfig) -> str:
+        col_names = [c.get_name() for c in self.__base_frame.columns()]
+        desired_columns = self.__get_desired_columns(col_names)
+
+        selected_columns = []
+        for col in desired_columns:
+            escaped = col.replace("\\", "\\\\").replace("'", "\\'")
+            selected_columns.append(f"'{escaped}'")
+        return f"{self.__base_frame.to_pure(config)}{config.separator(1)}->select(~[{', '.join(selected_columns)}])"
```

Comment:
Why is this logic so different from LegendQLApiSelectFunction once ____get_desired_columns method is defined?